okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -397,7 +398,10 @@ private boolean shouldRetry(IOException exception, Optional<Duration> backoff) {
                     // non-connect timeouts should not be retried
                     SocketTimeoutException socketTimeout = (SocketTimeoutException) exception;
                     if (socketTimeout.getMessage() == null
-                            || !socketTimeout.getMessage().contains("connect timed out")) {
+                            || !socketTimeout
+                                    .getMessage()
+                                    .toLowerCase(Locale.ENGLISH)
+                                    .contains("connect timed out")) {
```

Comment:
This is hilarious imho, in java 14 the message is `Connect timed out`

Reply:
Mind adding a comment describing both messages? Should we avoid toLowerCase String allocation in favor of matching “onnect timed out”?

Reply:
added a comment and removed tolowercase